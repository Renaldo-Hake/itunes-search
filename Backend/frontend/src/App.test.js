import React from 'react';
import renderer from 'react-test-renderer';
import SearchApi from './Components/ItunesSearch';


// Snapshot test for itunesSearch component
it('snapshot test', () => {
  const tree = renderer
    .create(<SearchApi/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


// testing itunes api inside itunesSearch
test('the data is Beyoncé', () => {
  const name = 'Beyoncé';
  return fetch(`https://itunes.apple.com/search?term=Beyonce`)
  .then(res => res.json())
  .then(data => {
    const response = data.results[1];
    expect(response.artistName).toBe(name);
  });
});