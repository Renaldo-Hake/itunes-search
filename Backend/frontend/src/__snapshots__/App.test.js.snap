// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Itunes"
>
  <form
    className=""
    id="formName"
  >
    <div>
      <h3
        id="title"
      >
        Itunes search bar
      </h3>
      <label
        className="form-label"
      >
        Search your favorite items from itunes:
      </label>
      <input
        className="form-control"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
      <div>
        <select
          id="mediaType"
          onChange={[Function]}
        >
          <option
            value="movie"
          >
            movie
          </option>
          <option
            value="podcast"
          >
            podcast
          </option>
          <option
            value="music"
          >
            music
          </option>
          <option
            value="audiobook"
          >
            audiobook
          </option>
          <option
            value="shortFilm"
          >
            short film
          </option>
          <option
            value="tvSeason"
          >
            TV show
          </option>
          <option
            value="software"
          >
            software
          </option>
          <option
            value="ebook"
          >
            ebook
          </option>
          <option
            value="all"
          >
            all
          </option>
        </select>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </form>
  <div
    className="itemsContainer"
  />
</div>
`;

exports[`snapshot test 1`] = `
<div
  className="Itunes"
>
  <form
    className=""
    id="formName"
  >
    <div>
      <h3
        id="title"
      >
        Itunes search bar
      </h3>
      <label
        className="form-label"
      >
        Search your favorite items from itunes:
      </label>
      <input
        className="form-control"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
      <div>
        <select
          id="mediaType"
          onChange={[Function]}
        >
          <option
            value="movie"
          >
            movie
          </option>
          <option
            value="podcast"
          >
            podcast
          </option>
          <option
            value="music"
          >
            music
          </option>
          <option
            value="audiobook"
          >
            audiobook
          </option>
          <option
            value="shortFilm"
          >
            short film
          </option>
          <option
            value="tvSeason"
          >
            TV show
          </option>
          <option
            value="software"
          >
            software
          </option>
          <option
            value="ebook"
          >
            ebook
          </option>
          <option
            value="all"
          >
            all
          </option>
        </select>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </form>
  <div
    className="itemsContainer"
  />
</div>
`;
